src/wasm/wasm.cpp

```diff
@@ -174,85 +172,82 @@ handleUnreachable(Block* block,
   }
 }
 
-void Block::finalize() {
-  if (list.size() == 0) {
-    type = Type::none;
-    return;
-  }
-  // The default type is what is at the end. Next we need to see if breaks and/
-  // or unreachability change that.
-  type = list.back()->type;
-  if (!name.is()) {
-    // Nothing branches here, so this is easy.
-    handleUnreachable(this, NoBreak);
-    return;
-  }
+void Block::finalize(std::optional<Type> type_, Breakability breakability) {
+  if (type_) {
+    type = *type_;
+    if (type == Type::none && list.size() > 0) {
+      handleUnreachable(this, breakability);
+    }
```

Comment:
Maybe early-return after this to avoid indenting all the rest in an else?

Reply:
Done. Also did the same for `If::finalize`. (Others (`Loop`, `Try`, and `TryTable`) were kinda small and didn't seem worth it)